frontend/src/concepts/pipelines/apiHooks/mlmd/useMlmdContext.ts

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import { MlmdContext, MlmdContextTypes } from '~/concepts/pipelines/apiHooks/mlmd/types';
+import { usePipelinesAPI } from '~/concepts/pipelines/context';
+import { GetContextByTypeAndNameRequest } from '~/third_party/mlmd';
+import useFetchState, {
+  FetchState,
+  FetchStateCallbackPromise,
+  NotReadyError,
+} from '~/utilities/useFetchState';
+
+/**
+ * A hook used to use the MLMD service and fetch the MLMD context
+ * If being used without name/type, this hook will throw an error
```

Comment:
I'm not sure why name and type are optional if the function doesn't work if they're not provided.

Reply:
I didn't test what would happen if they are not provided. This is also the logic from the KubeFlow codebase. I can test it now.

Reply:
Ok, I changed it to a nitpick, so maybe a no-op. Otherwise your changes lgtm. Feel free to squash your commits.

Reply:
Ok, it's just
```
request.setTypeName(type);
request.setContextName(name);
```
from the mlmd library doesn't allow the `undefined` to be set, not sure what will happen if I pass an empty string to it if it doesn't exist. But definitely, we always need to deal with the null value.